File: day1/js/day1_part1.js
import fs from 'fs';
import readline from 'readline';

var input_file = '../input.txt';
var r = readline.createInterface({
    input: fs.createReadStream(input_file)
});

let total = 0

r.on('line', function (text) {
    const lineWithoutLetters = text.replaceAll(/[a-z, A-Z]/g, '')
    if (!lineWithoutLetters.length) return
    else {
        total += parseInt(lineWithoutLetters[0] + lineWithoutLetters.slice(-1))
    }
}).on('close', () => {
    console.log(total);
})

File: day1/go/day1_part1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"regexp"
	"strconv"
)

func day1_part1() {
	readFile, err := os.Open("../input.txt")

	if err != nil {
		fmt.Println(err)
	}
	fileScanner := bufio.NewScanner(readFile)

	fileScanner.Split(bufio.ScanLines)

	total := 0
	for fileScanner.Scan() {

		// Remove all non-numeric characters
		reg := regexp.MustCompile("[a-z, A-Z]")
		stringWithoutLetters := reg.ReplaceAllString(fileScanner.Text(), "")

		if len(stringWithoutLetters) != 0 {
			// Construct the number
			number := fmt.Sprintf("%s%s", string(stringWithoutLetters[0]), string(stringWithoutLetters[len(stringWithoutLetters)-1]))

			// Convert to int and add to total
			res, err := strconv.Atoi(number)
			if err == nil {
				total += res
			}
		}

	}

	readFile.Close()
	fmt.Println("Part1: ", total)
}
